bridge/src/main/java/com/livefront/bridge/BridgeDelegate.java

```diff
@@ -341,4 +345,8 @@ private void writeToDisk(@NonNull String uuid,
                 .apply();
     }
 
+    private boolean openedFromDeeplink(Activity activity) {
+        return Intent.ACTION_VIEW.equals(activity.getIntent().getAction())
+                && activity.getIntent().getData() != null;
```

Comment:
Hmm...this seems much too generic to me to ensure this only handles the scenario you are talking about. For example, wouldn't this also be triggered from a deep link when the app is not in a backgrounded, cached state?

Reply:
Is that part of the argument...that maybe its OK to leave the state sometimes because _eventually_ you will launch the app in a fresh state?

Reply:
OK so I'd like to see if there is a much more generic solution here for detecting app launches that might contain saved state or not. I'm curious if one option here is to somehow hook into the [ActivityManager.getRunningTasks](https://developer.android.com/reference/android/app/ActivityManager.html#getRunningTasks(int)) to query for the [RunningTaskInfo](https://developer.android.com/reference/android/app/ActivityManager.RunningTaskInfo) and then check for the [numActivities](https://developer.android.com/reference/android/app/TaskInfo.html#numActivities) / [numRunning](https://developer.android.com/reference/android/app/ActivityManager.RunningTaskInfo.html#numRunning) (depending on the API version).

If that value is 1 and the `Activity` being passed in has no saved state, we'd clear data. If that value is greater than 1 and the `Activity` being passed in has no saved state, then we'd keep data around, since the rest of the "running" Activities in the stack _would_ have saved state. Now I'm not certain if that logic falls apart when there are different tasks running but its worth trying out I think.

Is this something you could experiment with?

Reply:
Good insight, I will find time to experiment on it, leave this pr, both of us could keep thinking about it.

Reply:
I did test and seems it could work, but seems those function will be deprecated and beside it's hard to pass a proper max stack size to `getRunningTasks` to get all the stack infos.

Reply:
Hmm...so I'm still willing to explore that direction. I'm aware of the current limitations of `ActivityManager.getRunningTasks` and its evolution over time and I agree that it's _possible_ that in some future Android version `ActivityManager.getRunningTasks` might not be a tool available to work with, but that's a risk I'm willing to take to make the logic safer for now. We could at least investigate [ActivityManager.getAppTasks()](https://developer.android.com/reference/android/app/ActivityManager.html#getAppTasks()) to see if that's slightly safer too. In the future they might provide something that replaces all this and improves upon it and if they don't we could also revisit your current solution. Would you be interested in at least pushing up a separate MR using that solution we could take a look at?

Reply:
ok, give me some time, I will make a separate pr for `getAppTasks`